### PR TITLE
NVG give you green vision

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
And general support for letting glasses tint your vision

http://i.imgur.com/dvJTIEl.png

Or I can change it to something less saturated like

http://i.imgur.com/ArEVDCd.png

### CODE REVIEW

**Comment by Supermichael777** [comment]
cool

**Comment by Xhuis** [comment]
I feel like it's _way_ too green. Otherwise, :+1:

**Comment by PKPenguin321** [comment]
Yeah, needs way less saturation

**Comment by Robustin** [comment]
For what purpose

**Comment by Durkel** [comment]
Didn't we do this with gasmasks and everyone hated it?

**Comment by SpaiR** [comment]
As you can see items in inventory get green colour too. To fix this problem you can use for example APPEARANCE_IGNORE_COLOUR or APPEARANCE_UI to make it more complex. This flag could be given at the moment, when item is picked up and removed when dropped, so you can easily implement it in right inventory procs. By the way you can use another decision.

**Comment by KazeEspada** [comment]
Make the green lighter in color and more clear.

**Comment by Dorsisdwarf** [comment]
Second is better.

R&D's one should be greener

**Comment by Lumipharon** [comment]
Neat.

Did that noir vision thing for the detective ever happen?

**Comment by GunHog** [comment]
Dets should get Noir glasses, but please do not do this to NVs, this would be incredibly annoying for Nuke Ops, who almost always wear a pair. :-1:

**Comment by Iamgoofball** [comment]
@KorPhaeron 
Do you think you could ripoff the MGS series "make enemies bright neon green" when wearing NVGs meme, and then dull the green on the NVGs vs tiles/objects?

**Comment by PKPenguin321** [comment]
>As you can see items in inventory get green colour too. To fix this problem you can use for example APPEARANCE_IGNORE_COLOUR

Are you telling me it's possible to have noirvision for the detective but excluding blood so it remains red????

**Comment by SpaiR** [comment]
> Are you telling me it's possible to have noirvision for the detective but excluding blood so it remains red????

I thought that this method is already well known. Paradise already use it https://github.com/ParadiseSS13/Paradise/pull/3676
And also they using matrix method for incolouring and it's much better then just colours, because matrices gives absolute control of tone, so it's very easy to find any palette you like.

**Comment by Bawhoppen** [comment]
can you make it just like a minor minor thing

**Comment by RemieRichards** [comment]
I don't like how you've done it in the back end (or more specifically, what you haven't done) but I'll do it at some point.

**Comment by kevinz000** [comment]
stop nerfing R&D

**Comment by CosmicScientist** [comment]
Thermal vision goggles to be like their real world counterparts when? (Well, one kind of them.)

**Comment by KorPhaeron** [comment]
Guess not then

File: code/modules/clothing/glasses/glasses.dm [hunk 1]
```diff
@@ -2,6 +2,20 @@
 	name = "glasses"
 	materials = list(MAT_GLASS = 250)
 
+/obj/item/clothing/glasses/dropped(mob/user)
+	remove_tint(user)
```
Comment (phil235) [inline code/modules/clothing/glasses/glasses.dm]: must call ..()

File: code/modules/clothing/glasses/glasses.dm [hunk 2]
```diff
@@ -2,6 +2,20 @@
...
 
+/obj/item/clothing/glasses/dropped(mob/user)
+	remove_tint(user)
+
+/obj/item/clothing/glasses/equipped(mob/user, slot)
+	if(slot != slot_glasses)
+		remove_tint(user)
+	else
+		if(user.client && !user.client.color)
+			user.client.color = color_tint
```
Comment (phil235) [inline code/modules/clothing/glasses/glasses.dm]: pretty sure equipped must always call ..() as well.

**Pull request closed without merging**